### ISSUE TITLE
M2.1.2 : Live site in production shows blank page with only 'Add Video' button.

### ISSUE THREAD

**Issue by kanduvisla** [magento/magento2#7195]
When I try to edit a product in production, I only get an 'Add Video'-button like this:

![screen shot 2016-10-26 at 14 23 06](https://cloud.githubusercontent.com/assets/346681/19725782/de187074-9b87-11e6-9cf5-7e9de0b6f920.png)

Source:

```


<div class="row">
    <div class="add-video-button-container">
        <button id="add_video_button"
                title="Add Video"
                data-role="add-video-button"
                type="button"
                class="action-secondary"
                data-ui-id="widget-button-1">
            <span>Add Video</span>
        </button>
    </div>
</div>


<div id="media_gallery_content"
     class="gallery"
     data-mage-init='{"openVideoModal":{}}'
     data-parent-component="product_form.product_form.block_gallery.block_gallery"
     data-images="[{&quot;value_id&quot;:&quot;1400&quot;,&quot;file&quot;:&quot;\/i\/f\/iflight-panther-1260.jpg&quot;,&quot;media_type&quot;:&quot;image&quot;,&quot;entity_id&quot;:&quot;970&quot;,&quot;label&quot;:null,&quot;position&quot;:&quot;1&quot;,&quot;disabled&quot;:&quot;0&quot;,&quot;label_default&quot;:null,&quot;position_default&quot;:&quot;1&quot;,&quot;disabled_default&quot;:&quot;0&quot;,&quot;video_provider&quot;:null,&quot;video_url&quot;:null,&quot;video_title&quot;:null,&quot;video_description&quot;:null,&quot;video_metadata&quot;:null,&quot;video_provider_default&quot;:null,&quot;video_url_default&quot;:null,&quot;video_title_default&quot;:null,&quot;video_description_default&quot;:null,&quot;video_metadata_default&quot;:null,&quot;url&quot;:&quot;http:\/\/dartshopper.hypernode.io:8880\/media\/catalog\/product\/i\/f\/iflight-panther-1260.jpg&quot;,&quot;size&quot;:221567}]"
     data-types="
```

... that's it. The source just ends there at `data-types="`.

I have no exception or anything useful in my `var/log`-folder. However, I do have a PHP error in my NGINX log: My PHP error in `/var/log/nginx/error.log` looks like this:

```
2016/10/26 12:17:01 [error] 1240#1240: 
*9590 FastCGI sent in stderr: "PHP message: PHP Fatal error: 
 Method Magento\Ui\TemplateEngine\Xhtml\Result::__toString() must not throw an exception, 
caught Error: Class 'select' not found in (...)/magento/module-ui/Component/Wrapper/UiComponent.php on line 0" while reading response header from upstream, 
client: 188.205.57.46, 
server: , 
request: "GET /admin/catalog/product/edit/id/966/key/7d7d7b0d476ea75c36e7ca4fcaa8d8a4570634031ebdfe81ba620619dda2a1b7/ HTTP/1.1", 
upstream: "fastcgi://127.0.0.1:9000", 
host: "xxx.hypernode.io:8880", 
referrer: "http://xxx.hypernode.io:8880/admin/catalog/product/index/key/b25ec23d81d96eac72d0fe2fba167d699f917a02f1807b58f9c0c39dee99575c/"
```

I tried the debug-tip on this site: http://magento.stackexchange.com/questions/120210/magento-2-impossible-to-trace-admin-grid-error-fatal-error-method-magento-ui-t
but the `getComponentName()`-method only returned `form`, so that's also not very helpful...

I tried clearing the cache, re-compiling, deleting `var/di`, all with no success.

Does anyone know what might be going on here?

**Comment by kanduvisla** [comment]
So far I'm trying to break down where this error occurs. It looks like the big hold-up is caused by a plugin in the Product Swatches module. The code seems to break when the interceptor tries to execute `\Magento\Swatches\Model\Plugin\Product::afterGetMediaAttributes()`.

The code breaks when the following line is executed in `Magento\Framework\Interception\Interceptor`:

```
// Call original method
$result = parent::$method(...array_values($arguments));
```

But still, I think the message `Class 'select' not found in (...)/vendor/magento/module-ui/Component/Wrapper/UiComponent.php on line 0` might also be a hint to the real problem.

**Comment by kanduvisla** [comment]
ARGH!

Sorry guys, my bad... turns out I screwed up the `eav_attribute`-table with a faulty upgradescript. Well at least, it was faulty at first. What it tried to do is create an attribute programatically with frontend input type `select`. But instead of setting `frontend_input` to `select`, I'd set `frontend` to select, causing Magento to look for a frontend model named `select`. Which doesn't exist, and which makes perfect sense now in the PHP error logged...

The irony is, I had already fixed this issue locally, so my local database was correct, but my production database had already created the faulty attribute on an earlier deployment.

So... closing this one. Hopes this story helps some other poor lad a couple of hours of debugging >:-(
